### PR TITLE
clipdel: add -F to match literal instead of regex

### PR DESCRIPTION
**Pull request by N-R-K**
this adds an -F flag similar to grep to allow matching literals instead of a regex.

useful in scripting context where the script has a literal line to match against. using -F avoids having to escape troublesome regex characters.

### CODE REVIEW

File: src/clipdel.c [hunk 1]
```diff
@@ -24,17 +25,24 @@ enum delete_mode {
...
  * be running DELETE_REAL.
  */
-static enum cs_remove_action _nonnull_
-remove_if_rgx_match(uint64_t hash _unused_, const char *line, void *private) {
+static enum cs_remove_action _nonnull_ remove_if_match(uint64_t hash _unused_,
+                                                       const char *line,
+                                                       void *private) {
     struct clipdel_state *state = private;
-    int ret = regexec(&state->rgx, line, 0, NULL, 0);
+    int ret = state->literal_match
```
Comment (cdown) [inline src/clipdel.c]: patch looks good, my only thought is that the nested ternary isn't the easiest to read. maybe just do it in two if branches? then you don't need to coerce to REG_NOMATCH, too.

**Comment by cdown** [comment]
Oh, and sorry, could you add a test to tests/x_integration_tests? :-)

**Comment by cdown** [comment]
Looks great, thank you!

**Comment by N-R-K** [comment]
I had to change the shebang to `/bin/bash` since `/usr/bin/bash` doesn't exist on my system, but then I noticed that I don't have `xsel` installed currently either. I added a test case, let's see if it passes the CI :)

**Comment by cdown** [comment]
oh, that path is wrong, it should be /usr/bin/env bash. let me fix that. thanks!

**Pull request merged**